scripts/proposals/reserveRestore.ts

```diff
@@ -0,0 +1,413 @@
+    "setMonthlyReputationDistribution(uint256)",
+    "setReserveRatioDailyExpansion(uint256,uint256)",
+    "setExecutionDailyLimit(uint256)", // set limit to 0 so old bridge cant be used
+    "addBlacklisted(address)",
+    "addBlacklisted(address)",
+    "addBlacklisted(address)",
+    "claimTokens(address,address)",
+    "upgradeTo(address)",
+    "upgradeTo(address)",
+    "upgradeTo(address)",
+    "upgradeTo(address)",
+    "upgradeTo(address)",
+    "upgradeTo(address)",
+    "upgradeTo(address)",
+    "setAddresses()",
+    "registerScheme(address,bytes32,bytes4,address)", // give upgrade contract permissions
+    "upgrade(address)",
+    "unpause()"
+  ];
+
+  const proposalFunctionInputs = [
+    ethers.utils.defaultAbiCoder.encode(["bool"], [true]),
+    ethers.utils.defaultAbiCoder.encode(["bool"], [true]),
+    ethers.utils.defaultAbiCoder.encode(["uint256"], [0]),
+    ethers.utils.defaultAbiCoder.encode(["uint256", "uint256"], [999711382710978, 1e15]),
+    ethers.utils.defaultAbiCoder.encode(["uint256"], [0]),
+    ethers.utils.defaultAbiCoder.encode(["address"], ["0xD17652350Cfd2A37bA2f947C910987a3B1A1c60d"]),
+    ethers.utils.defaultAbiCoder.encode(["address"], ["0xeC577447D314cf1e443e9f4488216651450DBE7c"]),
+    ethers.utils.defaultAbiCoder.encode(["address"], ["0x6738fA889fF31F82d9Fe8862ec025dbE318f3Fde"]),
+    ethers.utils.defaultAbiCoder.encode(["address", "address"], [release.GoodDollar, release.MpbBridge]),
+    ethers.utils.defaultAbiCoder.encode(["address"], [reserveImpl.address]),
+    ethers.utils.defaultAbiCoder.encode(["address"], [goodFundManagerImpl.address]),
+    ethers.utils.defaultAbiCoder.encode(["address"], [exchangeHelperImpl.address]),
+    ethers.utils.defaultAbiCoder.encode(["address"], [distHelperImplt.address]),
+    ethers.utils.defaultAbiCoder.encode(["address"], [stakersDistImpl.address]),
+    ethers.utils.defaultAbiCoder.encode(["address"], [marketMakerImpl.address]),
+    ethers.utils.defaultAbiCoder.encode(["address"], [govImpl.address]),
+    "0x", //setAddresses
+    ethers.utils.defaultAbiCoder.encode(
+      ["address", "bytes32", "bytes4", "address"],
+      [
+        upgradeImpl.address, //scheme
+        ethers.constants.HashZero, //paramshash
+        "0x000000f1", //permissions - minimal
+        release.Avatar
+      ]
+    ),
+    ethers.utils.defaultAbiCoder.encode(["address"], [funder]),
+    "0x"
+  ];
+
+  if (isProduction) {
+    await executeViaSafe(
+      proposalContracts,
+      proposalEthValues,
+      proposalFunctionSignatures,
+      proposalFunctionInputs,
+      protocolSettings.guardiansSafe,
+      "mainnet"
+    );
+  } else {
+    //simulation or dev envs
+    await executeViaGuardian(
+      proposalContracts,
+      proposalEthValues,
+      proposalFunctionSignatures,
+      proposalFunctionInputs,
+      guardian,
+      networkName
+    );
+  }
+
+  if (isSimulation) {
+    await mainnetPostChecks()
+  }
+};
+
+const mainnetPostChecks = async () => {
```

Comment:
it can only be done once the proposal is actually executed. it can not be run simply after the first production run.
the production run only prepares the transaction for the guardians.

Reply:
yes this checks will be run after the proposal is executed.
it will still be a "simulation" only on a fork post the proposal approval.

Reply:
yes, we can perform tests post deployment